index.js

```diff
@@ -43,20 +39,15 @@ function updateChannelsToHost(onError, serverCfg) {
     newChannels.forEach(channel => {
       if (!oldChannels.has(channel)) {
         serverCfg.channels.add(channel);
-        serverCfg.peer.join(channel, {}, err => {
-          if (!err) return;
-          if (onError) onError(err);
-          serverCfg.channels.delete(channel);
-          serverCfg.peer.leave(channel);
-        });
+        serverCfg.peer.join(crypto.createHash('sha256').update(channel).digest(), { lookup: true, announce: true });
       }
     });
 
     // oldChannels minus newChannels => leave
     oldChannels.forEach(channel => {
       if (!newChannels.has(channel)) {
         serverCfg.channels.delete(channel);
-        serverCfg.peer.leave(channel);
+        serverCfg.peer.leave(crypto.createHash('sha256').update(channel).digest());
```

Comment:
This is used quite some times, could you make a function `channelToBuf`?

Reply:
Can do.